### PR TITLE
[microTVM][Target]Consolidate `stm32` target to `micro` target

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR consolidates stm32 targets into micro targets and renames some of the STM micro targets with generalized name. In addition it adds -keys=arm_cpu,cpu by default to all ARM micro targets.

### CODE REVIEW

**Comment by mehrdadh** [comment]
cc @areusch for review

File: apps/microtvm/zephyr/template_project/boards.json [hunk 1]
```diff
@@ -43,15 +43,15 @@
     },
     "nucleo_f746zg": {
         "board": "nucleo_f746zg",
-        "model": "stm32f746xx",
+        "model": "stm32f7xx",
```
Comment (areusch) [inline apps/microtvm/zephyr/template_project/boards.json:46]: why these changes?

Comment (mehrdadh) [reply apps/microtvm/zephyr/template_project/boards.json:46]: `stm32f7xx` is more generic than `stm32f746xx`
https://www.st.com/en/microcontrollers-microprocessors/stm32f7-series.html

File: python/tvm/target/target.py [hunk 2]
```diff
@@ -443,22 +443,45 @@ def intel_graphics(model="unknown", options=None):
...
-    "nrf5340dk": ["-mcpu=cortex-m33"],
-    "rp2040": ["-mcpu=cortex-m0"],
-    "sam3x8e": ["-mcpu=cortex-m3"],
-    "stm32f746xx": ["-mcpu=cortex-m7", "-march=armv7e-m"],
-    "stm32h7xx": ["-mcpu=cortex-m7"],
-    "stm32l4r5zi": ["-mcpu=cortex-m4"],
-    "stm32u5xx": ["-mcpu=cortex-m33"],
-    "zynq_mp_r5": ["-mcpu=cortex-r5"],
+    # NXP
+    "imxrt10xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m7"],
```
Comment (janetsc) [inline python/tvm/target/target.py:477]: Does this also need "-march=armv7e-m"?

Comment (mehrdadh) [reply python/tvm/target/target.py:477]: I think `armv7e-m` should be on all cortex-m4 and cortex-m7 boards:
https://en.wikipedia.org/wiki/ARM_Cortex-M

Comment (mehrdadh) [reply python/tvm/target/target.py:477]: cc @areusch

Comment (areusch) [reply python/tvm/target/target.py:477]: i believe the compiler infers this from -mcpu for ARM parts. cc @Mousius if he knows more

Comment (mehrdadh) [reply python/tvm/target/target.py:477]: @areusch that was my expectation too, but I saw `-march=armv7e-m` in one of the targets and I got confused.

Comment (janetsc) [reply python/tvm/target/target.py:477]: +1 - that's why I asked, because I saw it specified above.  Maybe that can be removed?
